src/future.rs

```diff
@@ -75,26 +76,57 @@ enum RetryFuturePoll<A> where A: Action {
 
 /// Future that drives multiple attempts at an action via a retry strategy.
 pub struct Retry<I, A> where I: Iterator<Item=Duration>, A: Action {
+    retry_if: RetryIf<I, A, fn(&A::Error) -> bool>
+}
+
+impl<I, A> Retry<I, A> where I: Iterator<Item=Duration>, A: Action {
+    pub fn spawn<T: IntoIterator<IntoIter=I, Item=Duration>>(handle: Handle, strategy: T, action: A) -> Retry<I, A> {
+        fn retry_always<A: Action>(_error: &A::Error) -> bool { true };
+        Retry {
+            retry_if: RetryIf::with_condition(handle, strategy, action, retry_always::<A>)
```

Comment:
Have you tried making this `...(handle, strategy, action, |_| true as fn(&A::Error) -> bool)` here?

That might help getting rid of `retry_always`.

Reply:
Thanks for your suggestion. That works and looks much better...